### PR TITLE
CO-2975: Allow recipients to be specified via email - MERGE 10/8

### PR DESCRIPTION
**Pull request by davehensley**
### Why are you making this change? (required)

Core Objects is adding a feature to allow requests to the `/campaigns/trigger/send` and `/canvas/trigger/send` endpoints to specify the recipient(s) by email address (instead of by external ID or user alias).

### Related PRs, issues, or features (optional)

- [BD-3526](https://jira.braze.com/browse/BD-3526)
- [CO-2975](https://jira.braze.com/browse/CO-2975)

### Feature release date (optional)

- Most likely 8 October 2024

### Contributor checklist

- [ ] I confirm that my PR meets the following:
    - I signed the [Contribution License Agreement (CLA)](https://www.braze.com/docs/cla).
    - My style and voice follows the [Braze Style Guide](https://docs.google.com/document/u/2/d/e/2PACX-1vTluyDFO3ZEV7V6VvhXE4As_hSFwmnFFdU9g6_TrAYTgH1QmbRoEDDdn5GzKAB9vdBbIdyiFdoaJcNk/pub).
    - My content contains correct spelling and grammar.
    - All links are working correctly.
    - If I renamed or moved a file or directory, I set up [URL redirects](https://www.braze.com/docs/contributing/content_management/redirecting_urls) for each file.
    - If I updated or replaced an image, I did not remove the original image file from the repository. (For more information, see [Updating an image](https://www.braze.com/docs/contributing/content_management/images/#updating-an-image)).
    - If my PR is related to a paid SKU, third party, SMS, AI, or privacy, I have received written approval from Braze Legal.

### Submitting for review

If your PR meets the above requirements, select **Ready for review**, then add a reviewer:
- **Non-Braze contributors:** Add `braze-inc/docs-team` as the reviewer.
- **Braze employees:** Have any relevant subject matter experts (like engineers or product managers) review your work first, then add the [tech writer assigned to your specific vertical](https://confluence.braze.com/pages/viewpage.action?pageId=75039896). If you're not sure which tech writer to add, you can add `braze-inc/docs-team` instead. 

Thanks for contributing! We look forward to reading your work.

### CODE REVIEW

File: _docs/_api/endpoints/messaging/send_messages/post_send_triggered_campaigns.md [hunk 1]
```diff
@@ -83,6 +85,10 @@ Authorization: Bearer YOUR-REST-API-KEY
 - When `send_to_existing_only` is `true`, Braze will only send the message to existing users. However, this flag can't be used with user aliases. 
 - When `send_to_existing_only` is `false`, an attribute must be included. Braze will create a user with the `id` and attributes before sending the message.
 
++{% alert important %}
++Specifying a recipient via email address is currently in early access. Contact your customer success manager if you're interested in participating in this early access.
```
Comment (lydia-xie) [inline _docs/_api/endpoints/messaging/send_messages/post_send_triggered_campaigns.md:89]:
```suggestion
+Specifying a recipient by email address is currently in early access. Contact your customer success manager if you're interested in participating in this early access.
```

File: _docs/_api/endpoints/messaging/send_messages/post_send_triggered_canvases.md [hunk 2]
```diff
@@ -70,6 +72,10 @@ Authorization: Bearer YOUR-REST-API-KEY
 
 Customers using the API for server-to-server calls may need to allowlist the appropriate API URL if they're behind a firewall.
 
++{% alert important %}
++Specifying a recipient via email address is currently in early access. Contact your customer success manager if you're interested in participating in this early access.
```
Comment (lydia-xie) [inline _docs/_api/endpoints/messaging/send_messages/post_send_triggered_canvases.md:76]:
```suggestion
+Specifying a recipient by email address is currently in early access. Contact your customer success manager if you're interested in participating in this early access.
```

File: _docs/_api/objects_filters/recipient_object.md [hunk 3]
```diff
@@ -11,7 +11,11 @@ description: "This reference article explains the different components of the Br
 
 > The recipients object allows you to request or write information in our endpoints.
 
-Either `external_user_id` or `user_alias` is required in this object. **Requests must specify only one.**
+Either `external_user_id`, `user_alias`, or `email` is required in this object. **Requests must specify only one.**
+
+{% alert important %}
+Specifying a recipient via email address is currently in early access. Contact your customer success manager if you're interested in participating in this early access.
```
Comment (lydia-xie) [inline _docs/_api/objects_filters/recipient_object.md:17]:
```suggestion
Specifying a recipient by email address is currently in early access. Contact your customer success manager if you're interested in participating in this early access.
```

**Comment by davehensley** [comment]
> Hey @davehensley , this one is scheduled to go out today. Do you have reviewers that still need to sign off on this PR? Is it ready for the tech writing team to review?

Hi @josh-mccrowell-braze, it's ready for your team to review, and I've asked my team to take a quick look as well before it goes out. Thank you!

**Review by lydia-xie — APPROVED** [review]
lgtm! marking as approved and waiting for a review from the CO team

**Review by shihanxiong — APPROVED** [review]

**Pull request merged**